src/database/mod.rs

```
@@ -335,3 +354,55 @@ pub enum CoinType {
     Unspent,
     Spent,
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum LabelItem {
+    Address(bitcoin::Address),
+    Txid(bitcoin::Txid),
+    OutPoint(bitcoin::OutPoint),
+}
+
+impl From<bitcoin::Address> for LabelItem {
+    fn from(value: bitcoin::Address) -> Self {
+        Self::Address(value)
+    }
+}
+
+impl From<bitcoin::Txid> for LabelItem {
+    fn from(value: bitcoin::Txid) -> Self {
+        Self::Txid(value)
+    }
+}
+
+impl From<bitcoin::OutPoint> for LabelItem {
+    fn from(value: bitcoin::OutPoint) -> Self {
+        Self::OutPoint(value)
+    }
+}
+
+impl Display for LabelItem {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            LabelItem::Address(a) => write!(f, "{}", a),
+            LabelItem::Txid(a) => write!(f, "{}", a),
+            LabelItem::OutPoint(a) => write!(f, "{}", a),
+        }
+    }
+}
+
+impl FromStr for LabelItem {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<LabelItem, Self::Err> {
+        if let Ok(addr) = bitcoin::Address::from_str(s) {
+            let addr = addr.assume_checked();
```

No, we shouldn't do that. This may silently hide some errors.